src/controller/cve.controller/cve.controller.js

```diff
@@ -38,133 +18,99 @@ async function getCve (req, res) {
+      return res.status(200).json({ message: cveId + ' has been created.' }) // send message to user
+    })
 }
 
-// TODO: Request body sanitation and parameter validation
-// MongoDB: Updates a CVE if it exists. If no CVE exists, the CVE does not get created and no changes are made.
+// CPS: Updates a CVE if one exists for the specified CVE ID
+// MongoDB: Updates a CVE only if it exists. If no CVE exists, it does not create the CVE.
 async function updateCve (req, res) {
   const cve = req.body
   const cveId = req.params.id
-  const instance = axios.create({
-    timeout: 10000,
-    headers: {
-      'CVE-API-CNA': req.header('CVE-API-CNA'),
-      'CVE-API-SUBMITTER': req.header('CVE-API-SUBMITTER'),
-      'CVE-API-SECRET': req.header('CVE-API-SECRET')
-    }
-  })
-
-  try {
-    const cpsResult = await instance.post('https://cvecps-dev.mitre.org/cps/ws/cve/' + cveId + '/', cve)
-    const updatedCve = JSON.parse(cpsResult.data.Entry)
-
-    // CPS work item was approved
-    if (!cpsResult.data.Error) {
-      logger.info(cveId + ' has been updated in CPS')
-      logger.info(cpsResult.data.Message)
-
-      // Store CVE in MongoDB
-      Cve.findOneAndReplace()
-        .byCveId(cveId)
-        .replaceOne({ cve: updatedCve })
-        .exec((err, result) => {
-          if (err) {
-            logger.warn('An error ocurred in MongoDB. ' + cveId + ' was not updated in the database.')
-            logger.error(err.stack)
-          } else {
-            if (result.n === 0) {
-              logger.info(cveId + ' was not updated in MongoDB because it does not exists in the database.')
-            } else {
-              logger.info(cveId + ' was successfully updated in MongoDB')
-            }
-          }
 
-          // Respond to the user always
-          return res.status(200).json({ message: cveId + ' has been updated' }) // send message to user
-        })
-    } else {
-      logger.warn('The CPS work item was not approved. ' + cveId + ' was not updated.')
-      logger.warn(cpsResult.data.Error)
-      return res.status(500).json({ message: 'The CPS work item was not approved. ' + cveId + ' could not be updated.' }) // send error message to user
-    }
-  } catch (err) {
-    // Request to CPS failed
-    logger.error(err.stack)
-    return res.status(500).send('Internal Server Error') // send error message to user
+  if (cveId !== cve.CVE_data_meta.ID) {
+    return res.status(400).json({ message: cveId + ' parameter does not match ' + cve.CVE_data_meta.ID + ' in the JSON body.' })
   }
 
-  // res.status(200).json({ message: cveId + ' has been updated' })
+  const newCve = new Cve({ cve: cve, _id: false })
+
+  // Find and Update CVE in MongoDB
+  Cve.findOneAndReplace()
+    .byCveId(cveId)
+    .replaceOne(newCve)
+    .exec((err, result) => {
+      if (err) {
+        logger.warn('An error ocurred in MongoDB. ' + cveId + ' was not updated in the database.')
+        logger.error(err.stack)
```

Comment:
Include something like:

return res.status(500).json({ message: 'Failed to update ' + cveId + '.' })